canonical/k8s-snap
# Implement `dns` component and Components Refactor

## Overview
This Pull Request introduces a mechanism for separating logic for component management, allowing for the configuration of these components. Additionally, the DNS component is implemented.
## Changes
- Introduced custom handlers for the `dns` and `network` components.
- Refactored the disable logic.
- Refactored the `network` component to adhere to the new management logic.
- Updated CoreDNS to version 1.29.0.

## Manual Testing
- Pods
```
NAMESPACE     NAME                               READY   STATUS    RESTARTS   AGE
kube-system   cilium-operator-56bdb99ff6-j5w2q   1/1     Running   0          8m5s
kube-system   cilium-v4q8z                       1/1     Running   0          8m5s
kube-system   ck-dns-coredns-7f648ff498-z7zrj    1/1     Running   0          8m
```
- nslookup
```
root@k8s-test:~# /snap/k8s/current/bin/kubectl --kubeconfig /var/snap/k8s/common/etc/kubernetes/admin.conf run --rm -it --image alpine -- sh
If you don't see a command prompt, try pressing enter.
/ # nslookup canonical.com
Server:		10.152.183.10
Address:	10.152.183.10:53

Non-authoritative answer:
Name:	canonical.com
Address: 2620:2d:4000:1::28
Name:	canonical.com
Address: 2620:2d:4000:1::26
Name:	canonical.com
Address: 2620:2d:4000:1::27

Non-authoritative answer:
Name:	canonical.com
Address: 185.125.190.21
Name:	canonical.com
Address: 185.125.190.29
Name:	canonical.com
Address: 185.125.190.20

/ # nslookup google.com
Server:		10.152.183.10
Address:	10.152.183.10:53

Non-authoritative answer:
Name:	google.com
Address: 2800:3f0:4005:411::200e

Non-authoritative answer:
Name:	google.com
Address: 142.250.218.142
```
## Notes
- Testing will be introduced in a follow-up PR.
- Custom configuration will be introduced once the machinery to provide custom configuration is in place.
